src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../../types/Person';
+
+type Props = {
+  people: Person[],
+  setOnSelected: (person: Person) => void,
+  delay: number,
+};
+
+export const Dropdown: React.FC<Props> = React.memo(({
+  people,
+  setOnSelected,
+  delay,
+}) => {
+  const [value, setValue] = useState('');
+  const [appliedValue, setAppliedValue] = useState('');
+
+  const preparedPeople = useMemo(() => {
+    return people.filter(person => (
+      person.name.toLowerCase().includes(appliedValue)
+    )) || null;
+  }, [appliedValue]);
+
+  const applyValue = useCallback(
+    debounce(setAppliedValue, delay),
+    [delay],
+  );
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const normalisedEvent = event.target.value.toLowerCase();
+
+    setValue(normalisedEvent);
+    applyValue(normalisedEvent);
+  };
+
+  const handleClick = useCallback((
+    person: Person,
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+
+    setValue(person.name);
+    setOnSelected(person);
+    setAppliedValue('');
+  }, []);
```

Comment:
I would recommend moving this logic to App component (App component receives delay as props, while dropdown receives only memoized filteredPeople array and onSelect function), then your memoization will make sense. Here you rerender your component on every input change